src/assets/scripts/client/App.js

```diff
@@ -82,6 +76,22 @@ export default class App {
      * @param airportLoadList {array<object>}  List of airports to load
      */
     initiateDataLoad(airportLoadList, initialAirportToLoad) {
+        // Check if the json we try to load exists
+        // If it doesn't, return early. Otherwise the program hangs.
+        if ($.getJSON(`assets/airports/${initialAirportToLoad.toLowerCase()}.json`) == undefined) {
```

Comment:
I think you're starting down the right track here.  the problem with this implementation is you've now doubled the  network requests by making duplicate calls.

since we know that airlines and aircraft will always exist, we may be able to do:
```javascript
// within .done() 
const initialAirport = typeof airportResponse[0] !== 'undefined' ? airportResponse[0] : null;

this.setupChildren(
  airportLoadList,
  initialAirport,
  airlineResponse[0].airlines,
  aircraftResponse[0].aircraft
);
```

the only problem with that is I don't know if we hit the `.fail()` method if `initialAirportToLoad` comes back undefined.  might be something to try though?